src/main/groovy/com/bmuschko/gradle/clover/CloverPlugin.groovy

```diff
@@ -331,12 +331,12 @@ class CloverPlugin implements Plugin<Project> {
 
         @CompileDynamic
         private boolean isJavaSourceSetOf(SourceSet sourceSet, Test testTask) {
-            return testTask.ext.originalClasspath.contains(sourceSet.java.outputDir) && !testTask.originalTestClassesDir.contains(sourceSet.java.outputDir)
+            return testTask.ext.originalClasspath.contains(sourceSet.java.classesDirectory.get().asFile) && !testTask.originalTestClassesDir.contains(sourceSet.java.classesDirectory.get().asFile)
```

Comment:
Where does this `classesDirectory` property come from? Gradle's standard source set has `classesDirs`, which is a collection of files, not just one file.

Reply:
SourceDirectorySet is where this comes from

https://docs.gradle.org/current/dsl/org.gradle.api.file.SourceDirectorySet.html#org.gradle.api.file.SourceDirectorySet:classesDirectory